src/spatialdata/_core/_elements.py

```diff
@@ -0,0 +1,87 @@
+"""SpatialData elements."""
+from __future__ import annotations
+
+from collections import OrderedDict
+from typing import Any
+from warnings import warn
+
+from dask.dataframe.core import DataFrame as DaskDataFrame
+from datatree import DataTree
+from geopandas import GeoDataFrame
+
+from spatialdata._types import Raster_T
+from spatialdata._utils import multiscale_spatial_image_from_data_tree
+from spatialdata.models import (
+    Image2DModel,
+    Image3DModel,
+    Labels2DModel,
+    Labels3DModel,
+    PointsModel,
+    ShapesModel,
+    get_axes_names,
+    get_model,
+)
+
+
+class Elements(OrderedDict[str, Any]):
+    _shared_keys: set[str] = set()
+
+    def __setitem__(self, key: str, value: Any) -> None:
+        if key in self._shared_keys:
+            warn(f"Key `{key}` already exists.", UserWarning, stacklevel=2)
+        super().__setitem__(key, value)
+        self._shared_keys.add(key)
```

Comment:
```suggestion
class Elements(OrderedDict[str, Any]):

    def __init__(self):
        super().__init__()
        self._shared_keys = set()

    def __setitem__(self, key: str, value: Any) -> None:
        if key in self._shared_keys:
            warn(f"Key `{key}` already exists.", UserWarning, stacklevel=2)
        super().__setitem__(key, value)
        self._shared_keys.add(key)
```
This makes `_shared_keys` an instance attribute instead of a class attribute which ensures that we do not spam warnings when returning a filtered `SpatialData` object.

Reply:
Looks good to me.

Reply:
The suggestion from Wouter can't be applied because the code is outdated, but I also thing that `_shared_keys` (or `_existing_keys` if it gets renamed) should be an instance method and not a class method, because this could lead to wrong behaviors when multiple `SpatialData` objects are being handled at the same time.

Reply:
Ah, the latest code already uses instance attributes, so all good here.